src/video/video.c

```diff
@@ -169,12 +169,17 @@ static void video_set_blend_mode(object_array_blend_mode request_mode) {
 }
 
 void video_area_capture(surface *sur, int x, int y, int w, int h) {
+    // XXX hack: round up read width to nearest multiple of 4 for glReadPixels call
+    // could someone who knows GL better replace this magic number 4 with something from the API?
+    int bufw = (w + 3) / 4 * 4;
+
     render_target_activate(g_video_state.target);
-    unsigned char *buffer = omf_calloc(1, w * h);
-    glReadPixels(x, y, w, h, GL_RED, GL_UNSIGNED_BYTE, buffer);
-    surface_create_from_data_flip(sur, w, h, buffer);
+    unsigned char *buffer = omf_calloc(1, bufw * h);
+    glReadPixels(x, y, bufw, h, GL_RED, GL_UNSIGNED_BYTE, buffer);
+    surface_create_from_data_flip_stride(sur, w, h, bufw, buffer);
     surface_set_transparency(sur, -1);
     omf_free(buffer);
+    render_target_deactivate();
```

Comment:
render_target_deactivate() should only be called when we're preparing to render to the screen. Otherwise it's an unnecessary GL api call.